### PR TITLE
routes and context setup

### PR DESCRIPTION
**Pull request by Chiragmodi01**
- Setup routes and context
- Signup and login form
- Toggle theme

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -10,6 +10,25 @@
     overflow: hidden;
 }
 
+.App.light {
+    --primary: #ffffff;
+    --secondary: #f9f9f9;
+    --lightGrey: #e5e5e5;
+    --white: #121212;
+    /* --light-grey-scrollbar: #aaaaaa;
```
Comment (aurobindoGupta) [inline src/App.css:18]: remove commented

File: src/comps/Sidebar/Sidebar.css [hunk 2]
```diff
@@ -30,13 +30,25 @@
...
 }
+.sidebar-menu.active {
+    background-color: var(--lightGrey);
+}
 
 .Sidebar.tiny {
     display: none;
+}
+.menu-icon.active-history {
+    stroke-width: 30px;
```
Comment (aurobindoGupta) [inline src/comps/Sidebar/Sidebar.css]: rem?

File: src/comps/Sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -1,38 +1,44 @@
-import React from 'react'
+import React , { useState } from 'react'
 import './Sidebar.css'
 
-import {AiOutlineHome, AiOutlineClockCircle, AiOutlineHistory, RiThumbUpLine, MdOutlineVideoLibrary} from '../../utils/getIcons';
+import {AiFillHome, AiOutlineHome, AiFillClockCircle, AiOutlineClockCircle, AiOutlineHistory, RiThumbUpFill, RiThumbUpLine, MdVideoLibrary, MdOutlineVideoLibrary} from '../../utils/getIcons';
 import { useMain } from '../../helpers/context/main-context';
+import { NavLink, useLocation } from "react-router-dom";
 
 function Sidebar() {
```
Comment (aurobindoGupta) [inline src/comps/Sidebar/Sidebar.jsx:8]: preferably use es6

File: src/pages/Signup/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import {StyledForm} from '../../styledComps/index';
+import './Signup.css';
+import {Link} from 'react-router-dom';
+
+function Signup() {
```
Comment (aurobindoGupta) [inline src/pages/Signup/Signup.jsx:6]: use es6, will be useful later

**Review by aurobindoGupta — APPROVED** [review]
nice work, preferably use es6

**Review by harshitpaliwal95 — APPROVED** [review]
great UI Chirag good work

**Pull request merged**